### PR TITLE
Source Woocommerce: fixes

### PR DESCRIPTION
**Pull request by saimanoj**
## What
*Describe what the change is solving*
*Fixed many issues in the connector.*
 - *Store URL - `.com` is hardcoded in the connector. Since some stores have different extensions (`.in`, `.co.in`)*
 - *Start Date - it was pulling data from the store start date.*
 - *Headers - For some sources, woo-commerce API is expecting user-agent in headers.*
 - *Normalizations -  In order streams, some variable data type is not correct.*

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-woocommerce/integration_tests/configured_catalog.json [hunk 1]
```diff
@@ -932,4 +941,4 @@
       "destination_sync_mode": "append"
     }
   ]
-}
+}
```
Comment (harshithmullapudi) [inline airbyte-integrations/connectors/source-woocommerce/integration_tests/configured_catalog.json:944]: new line?

File: airbyte-integrations/connectors/source-woocommerce/integration_tests/configured_catalog.json [hunk 2]
```diff
@@ -385,7 +391,7 @@
                     "type": ["null", "integer"]
                   },
                   "quantity": {
-                    "type": ["null", "integer"]
+                    "type": ["null", "string"]
```
Comment (harshithmullapudi) [inline airbyte-integrations/connectors/source-woocommerce/integration_tests/configured_catalog.json:394]: Why did we change it to string?

File: airbyte-integrations/connectors/source-woocommerce/source_woocommerce/source.py [hunk 3]
```diff
@@ -148,11 +183,12 @@ def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) ->
         """
         shop = config["shop"]
         headers = {"Accept": "application/json"}
-        url = f"https://{shop}.com/wp-json/wc/v3/"
+        url = f"https://{shop}/wp-json/wc/v3/"
 
         try:
             auth = TokenAuthenticator(token=self._convert_auth_to_token(config["api_key"], config["api_secret"]), auth_method="Basic")
             headers = dict(Accept="application/json", **auth.get_auth_header())
+            headers['User-Agent']='Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.95 Safari/537.36'
```
Comment (harshithmullapudi) [inline airbyte-integrations/connectors/source-woocommerce/source_woocommerce/source.py:191]: can we declare this as a variable

File: airbyte-integrations/connectors/source-woocommerce/source_woocommerce/source.py [hunk 4]
```diff
@@ -75,6 +100,7 @@ def limit(self):
     state_checkpoint_interval = limit
     # Setting the default cursor field for all streams
     cursor_field = "date_modified"
+    range_days = 15  # date range is set to 15 days, because for conversion_window_days default value is 14. Range less than 15 days will break the integration tests.
```
Comment (harshithmullapudi) [inline airbyte-integrations/connectors/source-woocommerce/source_woocommerce/source.py:103]: where is this used?

File: airbyte-integrations/connectors/source-woocommerce/source_woocommerce/source.py [hunk 5]
```diff
@@ -29,19 +30,29 @@ class WoocommerceStream(HttpStream, ABC):
...
 
     def path(self, **kwargs) -> str:
         return f"{self.data_field}"
+        
+    def request_headers(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> Mapping[str, Any]:
+        #for some source user-agent is expected by the woo-commerce API
+        return {
+            'User-Agent' : 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.95 Safari/537.36',
```
Comment (harshithmullapudi) [inline airbyte-integrations/connectors/source-woocommerce/source_woocommerce/source.py:53]: declare this on the top

**Review by harshithmullapudi — CHANGES_REQUESTED** [review]
Can you run
1. `./gradlew format`
2. Run integration tests and share the screenshots

File: airbyte-integrations/connectors/source-woocommerce/integration_tests/configured_catalog.json (hunk 1 above)
Comment (saimanoj) [reply airbyte-integrations/connectors/source-woocommerce/integration_tests/configured_catalog.json:944]: yeah, added few columns in all streams

File: airbyte-integrations/connectors/source-woocommerce/integration_tests/configured_catalog.json (hunk 2 above)
Comment (saimanoj) [reply airbyte-integrations/connectors/source-woocommerce/integration_tests/configured_catalog.json:394]: Quantity is coming in float. so, it is failing the normalization

File: airbyte-integrations/connectors/source-woocommerce/source_woocommerce/source.py (hunk 4 above)
Comment (saimanoj) [reply airbyte-integrations/connectors/source-woocommerce/source_woocommerce/source.py]: no, we are not using it anywhere. Removed it

File: airbyte-integrations/connectors/source-woocommerce/integration_tests/configured_catalog.json (hunk 1 above)
Comment (harshithmullapudi) [reply airbyte-integrations/connectors/source-woocommerce/integration_tests/configured_catalog.json:944]: can you add a new line here

**Review by harshithmullapudi — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-woocommerce/integration_tests/configured_catalog.json (hunk 1 above)
Comment (saimanoj) [reply airbyte-integrations/connectors/source-woocommerce/integration_tests/configured_catalog.json]: added

**Comment by mihaly-dombi** [comment]
Hello everybody. Can I ask you guys to publish a new version with these changes to the docker hub? Thank you in advance.

**Comment by marcosmarxm** [comment]
@harshithmullapudi take a look here please

**Comment by harshithmullapudi** [comment]
@marcosmarxm I have published it here https://github.com/airbytehq/airbyte/pull/15645. But I will not merge the PR as that will run some tests I guess which can break the complete process. 

Any suggestions?

**Comment by marcosmarxm** [comment]
> But I will not merge the PR as that will run some tests I guess which can break the complete process.

Which processes?

**Comment by harshithmullapudi** [comment]
I think we run regularly tests for all connectors in source definition right?

**Pull request merged**